cowprotocol/cowswap
# Add feature flags to Cow Fortune

# Summary

* Adds new utilities to launch darkly setup;
  * `WithLDProvider`: This is a component that allows you to wrap a part of your tree with launch darkly. Used, so that we can move at which level we add launch darkly provider. Previous approach makes it available at `App` or below levels, which means any popup / updater etc. would not be able to access feature flags. This fixes that.
  * `FeatureGuard`: This is a component that takes the name of a feature flag, and only render if said feature flag is enabled. Added to simplify adding / removing feature flags (in this case, for cow fortune).
* Wrap Cow Fortune with a feature guard: `cowFortuneEnabled`.
  * This flag is currently enabled for *Test* and *Development* environments.

# To Test

1. Open Swap
2. Go to Launch Darkly
3. Find `cowFortuneEnabled`
4. Ensure you are on the correct environment
5. By flipping the switch you should be able to enable / disable cow fortune
